File: backend/app/schemas/region.py
from __future__ import annotations

from typing import Any

from pydantic import BaseModel


class RegionBounds(BaseModel):
    west: float
    south: float
    east: float
    north: float


class RegionResponse(BaseModel):
    regionId: str
    aoiName: str
    geojson: dict[str, Any]
    bounds: RegionBounds
    satelliteCollection: str
    predictionHorizonDays: int

File: frontend/src/types/region.ts
export interface RegionBounds {
  west: number;
  south: number;
  east: number;
  north: number;
}

export interface RegionInfo {
  regionId: string;
  aoiName: string;
  geojson: {
    type: string;
    coordinates: number[][][];
  };
  bounds: RegionBounds;
  satelliteCollection: string;
  predictionHorizonDays: number;
}
